scripts/bash/s3a_wrapper.sh

```diff
@@ -0,0 +1,174 @@
+
+set -e
+
+# Source env variables
+source "$(dirname "$0")/enceladus_env.sh"
+
+# Initialize variables
+keytab=""
+dataset_name=""
+dataset_version=""
+report_date=""
+report_version=""
+
+# The first argument is the name of the original script
+original_script="$(dirname "$0")/$(basename "$1")"
+
+# Shift the first argument so we can process the rest
+shift
+
+# Check if the original script exists
+if [[ ! -f "$original_script" ]]; then
+  echo "Error: The script '$original_script' does not exist in the current directory."
+  exit 1
+fi
+
+# Initialize an array to hold the other arguments
+other_args=()
+
+# Loop through arguments
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    --menas-auth-keytab)
+      keytab="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    --dataset-name)
+      dataset_name="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    --dataset-version)
+      dataset_version="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    --report-date)
+      report_date="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    --report-version)
+      report_version="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    *)    # unknown option
+      other_args+=("$1") # save it in an array for later
+      shift # past argument
+      ;;
+  esac
+done
+
+# Print the extracted variables
+echo "Keytab: $keytab"
+echo "Dataset Name: $dataset_name"
+echo "Dataset Version: $dataset_version"
+echo "Report Date: $report_date"
+echo "Report Version: $report_version"
+
+# Run klist to check for a current Kerberos ticket
+if klist -s; then
+    echo "Kerberos ticket found."
+else
+    echo "No Kerberos ticket found or ticket is expired. Please run kinit."
+fi
+
+# Get Dataset info
+response=$(curl --negotiate -s -u : "$MENAS_API/dataset/$dataset_name/$dataset_version")
+if [ $? -ne 0 ]; then
+    echo "Curl command failed."
+    exit 1
+fi
+
+# Parse the response using jq to extract hdfsPublishPath and hdfsPath
+hdfsPublishPath=$(echo "$response" | jq -r '.hdfsPublishPath')
+hdfsPath=$(echo "$response" | jq -r '.hdfsPath')
+
+# Check if the paths are null or not
+if [[ $hdfsPublishPath != "null" && $hdfsPath != "null" ]]; then
+    echo "hdfsPublishPath: $hdfsPublishPath"
+    echo "hdfsPath: $hdfsPath"
+else
+    echo "Could not find the required paths in the response."
+    exit 1
+fi
+
+# Run the original script with all the arguments
+"$original_script" "${other_args[@]}" \
+  --menas-auth-keytab "$keytab" \
+  --dataset-name "$dataset_name" \
+  --dataset-version "$dataset_version" \
+  --report-date "$report_date" \
+  --report-version "$report_version"
+
+# Save the exit code
+exit_code=$?
+
+# Run versions cleanup for publish on s3a
+if [[ $hdfsPublishPath == s3a://* ]]; then
+  echo "We have publish versions to clean:"
+  curl -X GET \
+    --header "x-api-key: $ECS_API_KEY" \
+    -d "{\"ecs_path\":\"${hdfsPublishPath#s3a://}\"}" \
```

Comment:
what is `${hdfsPublishPath#s3a://}`?
It seems a bit similar to  [${parameter/#pattern/string}](https://unix.stackexchange.com/questions/126087/what-does-the-hash-in-parameter-pattern-string-do), but not quite.

Reply:
```bash
$> hdfsPublishPath="s3a://some_bucket/my/path/there"
$> echo ${hdfsPublishPath#s3a://}
some_bucket/my/path/there
```

It strips the s3a prefix from the path saved in HDFS so it can be used by the mapping api